File: CSES/Mathematics/q24.cpp
n=int(input())
ans=0.0
arr=[]

arr = list(map(int,input().split()))
  
for i in range(n):
  for j in range(i+1,n):
    ri=arr[i]
    rj=arr[j]
    if ri<=rj:
      val=ri-1
      ans+=val/(2.0*rj)
    else:
      val=2*ri-rj-1
      ans+=val/(2.0*ri)

print("{:.6f}".format(ans))
 

File: CSES/Mathematics/q24.py
n,k=map(int, input().split())
ans=0.0

def binexp(a,n):
  res=1;
  while n>0:
    if n&1: 
      res*=a
    n>>=1
    a*=a
  return res
  
for i in range(1,k+1):
  ans+=i*binexp(i/k,n)
  ans-=i*binexp((i-1.0)/k,n)  

print("{:.6f}".format(ans))
 

